bgc_data/nrs_derived_indices_data.sql

```diff
@@ -0,0 +1,283 @@
+-- Materialized view for the NRS Derived Indices product
+-- To be served as a WFS layer by Geoserver
+CREATE MATERIALIZED VIEW nrs_derived_indices_data AS
+WITH
+
+-- Zooplankton temp tables ------------------------------------------------------------------------
+zoopinfo_mod AS (
+    -- reclassify diet column in zoopinfo
+    SELECT taxon_name,
+           length_mm,
+           CASE WHEN zi.diet = 'Carnivore' THEN 'CC'
+                WHEN zi.diet = 'Omnivore' THEN 'CO'
+                WHEN zi.diet = 'Herbivore' THEN 'CO'
+               END AS diet
+    FROM zoopinfo zi
+),
+zoop_by_trip AS (
+    -- per-trip stats for all zooplankton
+    SELECT trip_code,
+           sum(zoop_abundance_m3) AS "ZoopAbundance_m3"
+    FROM bgc_zoop_raw
+    GROUP BY trip_code
+),
+copepods_by_trip AS (
+    -- per-trip stats for all copepods
+    SELECT zr.trip_code,
+           sum(zr.zoop_abundance_m3) AS "CopeAbundance_m3",
+           sum(zr.zoop_abundance_m3 * zi.length_mm) AS "CopeSumAbundanceLength",
+           sum(zr.zoop_abundance_m3) FILTER ( WHERE zi.diet = 'CC') AS "CC",
+           sum(zr.zoop_abundance_m3) FILTER ( WHERE zi.diet = 'CO') AS "CO"
+    FROM bgc_zoop_raw zr LEFT JOIN zoopinfo_mod zi USING (taxon_name)
+    WHERE zr.copepod = 'COPEPOD'
+    GROUP BY trip_code
+),
+copepod_species_by_trip_species AS (
+    -- zooplankton raw data filtered to include only correctly identified copepod species
+    -- taxon_name updated to identify each unique species
+    -- grouped by trip_code & taxon_name
+    SELECT trip_code,
+           genus || ' ' || substring(species, '^\w+') AS taxon_name,
+           sum(taxon_count) AS taxon_count
+    FROM bgc_zoop_raw
+    WHERE copepod = 'COPEPOD' AND
+          species != 'spp.' AND
+          species NOT LIKE '%cf.%' AND
+          species NOT LIKE '%/%' AND
+          species NOT LIKE '%grp%'
+    GROUP BY trip_code, genus || ' ' || substring(species, '^\w+')
+),
+copepod_species_by_trip AS (
+    -- count number of copepod species
+    SELECT trip_code,
+           count(*) AS "NoCopepodSpecies_Sample",
+           sum(taxon_count) AS total_taxon_count,
+           sum(taxon_count * ln(taxon_count)) AS total_n_logn
+    FROM copepod_species_by_trip_species
+    GROUP BY trip_code
+),
+
+-- Phytoplankton temp tables ----------------------------------------------------------------------
+phyto_filtered AS (
+    -- filter out taxon_group "Other" from raw phyto data
+    -- create genus_species field for counting valid species
+    -- convert biovolume of cells to Carbon based on taxon_group
+    SELECT *,
+           CASE WHEN (species = 'spp.' OR
+                      species LIKE '%cf.%' OR
+                      species LIKE '%/%' OR
+                      species LIKE '%grp%') THEN NULL
```

Comment:
```suggestion
                      species LIKE '%complex%' OR
                      species NOT LIKE '%type%' OR
                      species NOT LIKE '%cyst%' ) THEN NULL
```
Same as earlier, I think these are the filters agreed with Claire for phytoplankton.

Reply:
Ok, but here we're matching rows to _exclude_, so it should be `species LIKE '%type%'` instead of `NOT LIKE`
Fixed in last commit

Reply:
correct!